Simplify Pmd custom rules configuration (#22211)
gradle/gradle: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

}
```

==== Simplifying PMD custom rules configuration

> Previously to configure PMD to use custom rules and ignore the default ones you have to explicitly clear default rules with following line `ruleSets = []`.
> In the Gradle 8.0 it's done by default as soon as `ruleSetConfig` or `ruleSetFiles` was set to non-empty value.

Reviewer: ```suggestion
Previously, you had to explicitly configure PMD to ignore default rules with `ruleSets = []`.
In the Gradle 8.0, setting `ruleSetConfig` or `ruleSetFiles` to a non-empty value implicitly ignores default rules.
```
Suggestion: trim some words, wording consistency.
Consider defining the PMD acronym in the first sentence with "Pxxxx Mxxxx Dxxxxx (PMD)", and possibly provide a link to other documentation on the subject.

Author: @nathan-contino Thanks!  Funny thing is PMD documentation doesn't clarifying what PMD acronym stands for:)
